src/modules/core/components/UserPickerWithSearch/hooks.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useRef } from 'react';
+
+import { FIX_TRIGGER_EVENT_NAME } from '~pages/ExpenditurePage/constants';
+
+const useUserTriggerFocus = (
+  index: number,
+  disabled: boolean | undefined,
+  toggleDropdown: () => void,
+) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    let timeout: ReturnType<typeof setTimeout>;
+
+    const fixAction = (e) => {
+      const {
+        detail: { order },
+      } = e;
+      // isDesiredEl tells to trigger only needed element and not for example all elements in the listing
+      const isDesiredEl = order === index;
```

Comment:
Would it be possible to also target the elements by name here? Just like in the DatePicker version of this hook.